### PR TITLE
Facelift of *_file functions

### PR DESCRIPTION
**Pull request by karel-m**

### CODE REVIEW

**Comment by karel-m** [comment]
Cosmetic observation - the order of params (key, filename vs. filename, key) varies a bit by different MACs:
```
hmac_file(hashid, fname, key, 32, buf, &len);
omac_file(cipherid, key, 32, fname, buf, &len);
pmac_file(cipherid, key, 32, fname, buf, &len);
xcbc_file(cipherid, key, 32, fname, buf, &len);
f9_file(cipherid, key, 32, fname, buf, &len);
poly1305_file(fname, key, 32, buf, &len);
```

File: src/mac/hmac/hmac_file.c [hunk 1]
```diff
@@ -63,12 +59,15 @@ int hmac_file(int hash, const char *fname,
       goto LBL_ERR;
    }
 
-   /* process the file contents */
+   if ((err = hmac_init(&hmac, hash, key, keylen)) != CRYPT_OK) {
```
Comment (sjaeckel) [inline src/mac/hmac/hmac_file.c]: why this? I liked it better before

... even if that means modifying all the others... :)

File: src/mac/hmac/hmac_file.c [hunk 2]
```diff
@@ -45,49 +45,51 @@ int hmac_file(int hash, const char *fname,
...
+   err = hmac_done(&hmac, out, outlen);
 
+LBL_ERR:
 #ifdef LTC_CLEAN_STACK
-   /* clear memory */
-   zeromem(buf, sizeof(buf));
+   zeromem(&hmac, sizeof(hmac_state));
 #endif
-   return CRYPT_OK;
+   XFREE(buf);
```
Comment (sjaeckel) [inline src/mac/hmac/hmac_file.c]: shouldn't we zero `buf` as well?

**Comment by sjaeckel** [comment]
> Cosmetic observation - the order of params (key, filename vs. filename, key) varies a bit by different MACs:
> ```
> hmac_file(hashid, fname, key, 32, buf, &len);
> omac_file(cipherid, key, 32, fname, buf, &len);
> pmac_file(cipherid, key, 32, fname, buf, &len);
> xcbc_file(cipherid, key, 32, fname, buf, &len);
> f9_file(cipherid, key, 32, fname, buf, &len);
> poly1305_file(fname, key, 32, buf, &len);
> ```

choose whatever you think suits best, is not that much work (also in regards of API changes of existing functions) and fits in the pattern `XX_file(<inputs>, <outputs>)` :)

File: src/mac/hmac/hmac_file.c (hunk 2 above)
Comment (karel-m) [inline src/mac/hmac/hmac_file.c]: we should

File: src/mac/hmac/hmac_file.c (hunk 1 above)
Comment (karel-m) [inline src/mac/hmac/hmac_file.c]: you mean moving `hmac_init` before calling `fopen`?

Comment (sjaeckel) [inline src/mac/hmac/hmac_file.c]: yep

File: src/mac/poly1305/poly1305_file.c [hunk 3]
```diff
@@ -18,14 +18,14 @@
 
 /**
   POLY1305 a file
-  @param fname    The name of the file you wish to POLY1305
   @param key      The secret key
   @param keylen   The length of the secret key
+  @param fname    The name of the file you wish to POLY1305
   @param out      [out] The POLY1305 authentication tag
```
Comment (sjaeckel) [inline src/mac/poly1305/poly1305_file.c]: s/out/mac

**Comment by karel-m** [comment]
@sjaeckel is this PR ok for merging?

**Comment by karel-m** [comment]
Now it should be back to:
```
hmac_file(hashid, fname, key, 32, buf, &len);
omac_file(cipherid, key, 32, fname, buf, &len);
pmac_file(cipherid, key, 32, fname, buf, &len);
xcbc_file(cipherid, key, 32, fname, buf, &len);
f9_file(cipherid, key, 32, fname, buf, &len);
poly1305_file(fname, key, 32, buf, &len);
```

**Comment by karel-m** [comment]
rebased and reorganised into 3 commits

**Pull request merged**